File: guests/python/styling.py
"""The styling conformance scene (tools/scenes/styling.steps): brand accent,
role tier, window inset."""

import sys

import kaya

app = kaya.App()


def delete():
    status.set("deleted")


def save():
    status.set("saved")


with app.window(title="styling", width=480.0, height=360.0, inset=0.0):
    # BEFORE THE FIRST MOUNT: the scope mounts on exit.
    kaya.brand_accent(0x3584E4)
    heading = kaya.signal("Sections")
    status = kaya.signal("ready")
    with kaya.column():
        # expect_ax resolves a target through its AUTHORED id.
        kaya.heading(bind=heading).a11y_id("title")  # label#0
        kaya.label(bind=status)  # label#1
        kaya.button("Delete", on_click=delete).role(
            kaya.Role.DESTRUCTIVE).a11y_id("delete")  # button#0
        kaya.button("Save", on_click=save).role(
            kaya.Role.PROMINENT).a11y_id("save")  # button#1
        # Declared so every backend's caption arm runs: no AX observable.
        kaya.caption("captioned")  # label#2

sys.exit(app.run())

File: guests/js/styling.ts
// The styling conformance scene (tools/scenes/styling.steps): brand accent,
// role tier, window inset. `Role.HEADING` is the one role a lane can read.

import * as kaya from "kaya-gui";

const app = new kaya.App();

let status!: kaya.Signal<string>;

function onDelete(): void {
  status.set("deleted");
}

function onSave(): void {
  status.set("saved");
}

app.window({ title: "styling", width: 480, height: 360, inset: 0 }, () => {
  // BEFORE THE FIRST MOUNT: the scope mounts on exit.
  kaya.brandAccent(0x3584e4);
  const heading = kaya.signal("Sections");
  status = kaya.signal("ready");
  kaya.column(() => {
    // expect_ax resolves a target through its AUTHORED id.
    kaya.heading({ bind: heading }).a11yId("title"); // label#0
    kaya.label({ bind: status }); // label#1
    kaya.button("Delete", { onClick: onDelete }).role(kaya.Role.DESTRUCTIVE).a11yId("delete"); // button#0
    kaya.button("Save", { onClick: onSave }).role(kaya.Role.PROMINENT).a11yId("save"); // button#1
    // Declared so every backend's caption arm runs: no AX observable.
    kaya.caption("captioned"); // label#2
  });
});

app.run();
